libraries/apollo-normalized-cache-api-incubating/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/api/internal/OptimisticNormalizedCacheWrapper.kt

```diff
@@ -30,49 +29,39 @@ class OptimisticCache : NormalizedCache() {
 
   override fun loadRecords(keys: Collection<String>, cacheHeaders: CacheHeaders): Collection<Record> {
     return lock.read {
-      val nonOptimisticRecords = nextCache?.loadRecords(keys, cacheHeaders)?.associateBy { it.key } ?: emptyMap()
+      val nonOptimisticRecords = wrapped.loadRecords(keys, cacheHeaders).associateBy { it.key }
       keys.mapNotNull { key ->
         nonOptimisticRecords[key].mergeJournalRecord(key)
       }
     }
   }
 
-  override fun merge(record: Record, cacheHeaders: CacheHeaders): Set<String> {
-    return lock.write { nextCache?.merge(record, cacheHeaders) } ?: emptySet()
-  }
-
-  override fun merge(records: Collection<Record>, cacheHeaders: CacheHeaders): Set<String> {
-    return lock.write { nextCache?.merge(records, cacheHeaders) } ?: emptySet()
-  }
-
-  @ApolloExperimental
   override fun merge(record: Record, cacheHeaders: CacheHeaders, recordMerger: RecordMerger): Set<String> {
-    return lock.write { nextCache?.merge(record, cacheHeaders, recordMerger) } ?: emptySet()
+    return lock.write { wrapped.merge(record, cacheHeaders, recordMerger) }
   }
 
-  @ApolloExperimental
   override fun merge(records: Collection<Record>, cacheHeaders: CacheHeaders, recordMerger: RecordMerger): Set<String> {
-    return lock.write { nextCache?.merge(records, cacheHeaders, recordMerger) } ?: emptySet()
+    return lock.write { wrapped.merge(records, cacheHeaders, recordMerger) }
   }
 
   override fun clearAll() {
     lock.write {
       recordJournals.clear()
-      nextCache?.clearAll()
+      wrapped.clearAll()
     }
   }
 
   override fun remove(cacheKey: CacheKey, cascade: Boolean): Boolean {
     return lock.write {
-      var result: Boolean = nextCache?.remove(cacheKey, cascade) ?: false
+      var result: Boolean = wrapped.remove(cacheKey, cascade)
 
       val recordJournal = recordJournals[cacheKey.key]
       if (recordJournal != null) {
         recordJournals.remove(cacheKey.key)
         result = true
         if (cascade) {
           for (cacheReference in recordJournal.current.referencedFields()) {
-            result = result && remove(CacheKey(cacheReference.key), true)
+            result = result || remove(CacheKey(cacheReference.key), true)
```

Comment:
I was going to say this looks like it needs to be backported to non-incubating but sounds like the whole semantics of the return value need to be refined here, right?

Reply:
yeah the `&&` didn't make much sense to me but if so the doc should also be updated to be precise:
```diff
- * @return `true` if a record with such key was successfully removed, `false` otherwise
+ * @return `true` if at least one record was successfully removed, `false` otherwise
```

Reply:
Not yet - I could open a dedicated ticket